src/middleware.js

```diff
@@ -49,15 +49,9 @@ export default function wrapper(context) {
         return;
       }
 
-      try {
-        content = context.outputFileSystem.readFileSync(filename);
-      } catch (_ignoreError) {
-        await goNext();
-        return;
-      }
-
-      // Buffer
-      content = handleRangeHeaders(content, req, res);
+      // Content-Type and headers need to be set before checking if
+      // the file is in the outputFileSystem, as these things should be
+      // applied to all files that are being served
```

Comment:
correct me if this approach is incorrect, but it seems that headers and content types need to be applied before rejecting a file that is not in the outputFileSystem. This is because files/routes served by webpack-dev-server like `localhost:8080/main`, or files that are in the user's directory and are being served, are not in the middleware filesystem but should still have the headers and content types applied to them. That is what the previous behavior was. But let me know if this was an intentional breaking change.

Reply:
Yes, looks like regression